Sources/PointFree/Routes.swift

```diff
@@ -86,11 +90,47 @@ private func redirectUnrelatedDomains<A>(
         } else {
           return nil
         }
-      }
-      ?? middleware(conn)
+        }
+        ?? middleware(conn)
+    }
+}
+
+// TODO: move to HttpPipeline
+private func requireHttps<A>(
+  _ middleware: @escaping Middleware<StatusLineOpen, ResponseEnded, A, Data?>
+  )
+  -> Middleware<StatusLineOpen, ResponseEnded, A, Data?> {
+
+    return { conn in
+      conn.request.url
+        .filterOptional { $0.scheme == .some("http") }
+        .flatMap(makeHttps)
+        .map {
+          conn
+            |> writeStatus(.movedPermanently)
+            |> writeHeader(.location($0.absoluteString))
+            |> map(const(nil))
+            |> closeHeaders
+            |> end
+        }
+        ?? middleware(conn)
     }
 }
 
+// TODO: move to httppipeline?
+private func makeHttps(url: URL) -> URL? {
+  var components = URLComponents(url: url, resolvingAgainstBaseURL: false)
+  components?.scheme = "https"
```

Comment:
This'd be fun to lens/prism about.

Reply:
oh jeez, totally! check it here! https://github.com/pointfreeco/pointfreeco/pull/7/commits/30f8d780f38eeed001bfcce168c59fe1b6504538